### PR TITLE
end my eternal suffering

### PR DESCRIPTION
**Pull request by MoriyaShiine**

### CODE REVIEW

**Comment by williewillus** [comment]
I only see generated data changes, but did you change the generator code itself? Is that committed?

**Comment by MoriyaShiine** [comment]
no

**Comment by williewillus** [comment]
um, what do you mean.

jsons in `src/generated` should NEVER be edited manually. Fix the data gen code then regenerate the jsons.

**Comment by MoriyaShiine** [comment]
sad

File: src/main/java/vazkii/botania/common/lib/ModTags.java [hunk 1]
```diff
@@ -184,10 +184,10 @@
...
 		public static final Tag.Identified<Block> LIVINGWOOD = tag("livingwood");
 
-		public static final Tag.Identified<Block> BLOCKS_ELEMENTIUM = forgeTag("storage_blocks/elementium");
-		public static final Tag.Identified<Block> BLOCKS_MANASTEEL = forgeTag("storage_blocks/manasteel");
-		public static final Tag.Identified<Block> BLOCKS_QUARTZ = forgeTag("storage_blocks/quartz");
-		public static final Tag.Identified<Block> BLOCKS_TERRASTEEL = forgeTag("storage_blocks/terrasteel");
+		public static final Tag.Identified<Block> BLOCKS_ELEMENTIUM = tag("elementium_blocks");
+		public static final Tag.Identified<Block> BLOCKS_MANASTEEL = tag("manasteel_blocks");
+		public static final Tag.Identified<Block> BLOCKS_QUARTZ = commonTag("quartz_blocks");
+		public static final Tag.Identified<Block> BLOCKS_TERRASTEEL = tag("terrasteel_blocks");
```
Comment (williewillus) [inline src/main/java/vazkii/botania/common/lib/ModTags.java:190]: so is it commonTag or tag? You're mixing both here.

File: src/main/java/vazkii/botania/data/ItemModelProvider.java [hunk 2]
```diff
@@ -1,410 +1,437 @@
-/*
```
Comment (williewillus) [inline src/main/java/vazkii/botania/data/ItemModelProvider.java:1]: revert this please.

File: src/main/java/vazkii/botania/data/ItemTagProvider.java [hunk 3]
```diff
@@ -79,32 +79,17 @@ protected void configure() {
...
-		*/
 
 		this.getOrCreateTagBuilder(ModTags.Items.NUGGETS_ELEMENTIUM).add(ModItems.elementiumNugget);
 		this.getOrCreateTagBuilder(ModTags.Items.NUGGETS_MANASTEEL).add(ModItems.manasteelNugget);
 		this.getOrCreateTagBuilder(ModTags.Items.NUGGETS_TERRASTEEL).add(ModItems.terrasteelNugget);
-		/* todo 1.16-fabric
-		this.getOrCreateTagBuilder(Tags.Items.NUGGETS).addTag(ModTags.Items.NUGGETS_ELEMENTIUM);
-		this.getOrCreateTagBuilder(Tags.Items.NUGGETS).addTag(ModTags.Items.NUGGETS_MANASTEEL);
-		this.getOrCreateTagBuilder(Tags.Items.NUGGETS).addTag(ModTags.Items.NUGGETS_TERRASTEEL);
-		*/
```
Comment (williewillus) [inline src/main/java/vazkii/botania/data/ItemTagProvider.java:107]: are these being reimplemented?

Comment (MoriyaShiine) [reply src/main/java/vazkii/botania/data/ItemTagProvider.java:107]: no, there are no common tags for generic storage blocks

File: src/main/java/vazkii/botania/common/lib/ModTags.java (hunk 1 above)
Comment (MoriyaShiine) [reply src/main/java/vazkii/botania/common/lib/ModTags.java:190]: there's a common tag for quartz, mods that add their own should have it in their own namespace

File: src/main/java/vazkii/botania/data/ItemModelProvider.java (hunk 2 above)
Comment (MoriyaShiine) [reply src/main/java/vazkii/botania/data/ItemModelProvider.java]: done, was an accident lol

File: src/main/java/vazkii/botania/data/BlockTagProvider.java [hunk 4]
```diff
@@ -129,8 +123,10 @@ protected void configure() {
 				ModBlocks.corporeaBrickWall, ModBlocks.corporeaCrystalCube, ModBlocks.corporeaFunnel, ModBlocks.corporeaIndex,
 				ModBlocks.corporeaInterceptor, ModBlocks.corporeaSlab, ModBlocks.corporeaStairs);
 
-//		getOrCreateTagBuilder(ModTags.Blocks.TERRAFORMABLE).addTag(Tags.Blocks.STONE).addTag(Tags.Blocks.DIRT).addTag(BlockTags.SAND)
-//				.add(Blocks.GRASS_BLOCK, Blocks.GRAVEL, Blocks.SNOW); TODO work around forge bug or wait for fix?
+		getOrCreateTagBuilder(ModTags.Blocks.TERRAFORMABLE).add(Blocks.ANDESITE, Blocks.DIORITE, Blocks.GRANITE, Blocks.INFESTED_STONE, Blocks.STONE, Blocks.POLISHED_ANDESITE, Blocks.POLISHED_DIORITE, Blocks.POLISHED_GRANITE)
+				.add(Blocks.DIRT, Blocks.GRASS_BLOCK, Blocks.COARSE_DIRT, Blocks.PODZOL, Blocks.MYCELIUM)
+				.add(Blocks.GRASS_BLOCK, Blocks.GRAVEL, Blocks.SNOW)
+				.add(Blocks.SAND, Blocks.RED_SAND); // #minecraft:sand causes an error for some reason
```
Comment (williewillus) [inline src/main/java/vazkii/botania/data/BlockTagProvider.java:129]: can you explain?

Comment (MoriyaShiine) [reply src/main/java/vazkii/botania/data/BlockTagProvider.java:129]: it contains all of the blocks that were previously in forge tags that don't exist in fabric

Comment (williewillus) [reply src/main/java/vazkii/botania/data/BlockTagProvider.java:129]: no, the "causes an error" part. What error? Did you try fixing it?

Comment (MoriyaShiine) [reply src/main/java/vazkii/botania/data/BlockTagProvider.java:129]: trying to do .addTag(BlockTags.SAND) causes the command to fail with "An unexpected error occurred trying to execute that command." It appears to fail with any block tag and I'm not sure why, so I just put the blocks in the sand tag instead

Comment (williewillus) [reply src/main/java/vazkii/botania/data/BlockTagProvider.java:129]: catch the exception and log it to see what it is. we should try to do things right.

**Pull request merged**